File: legacy/java/tests/StrataCommonTest/src/strata1/common/task/TestTask.java
// ##########################################################################
// # File Name:	TestTask.java
// #
// # Copyright:	2014, Sapientia Systems, LLC. All Rights Reserved.
// #
// # License:	This file is part of the StrataCommonTest Framework.
// #
// #   			The StrataCommonTest Framework is free software: you 
// #			can redistribute it and/or modify it under the terms of 
// #			the GNU Lesser General Public License as published by
// #    		the Free Software Foundation, either version 3 of the 
// #			License, or (at your option) any later version.
// #
// #    		The StrataCommonTest Framework is distributed in the 
// #			hope that it will be useful, but WITHOUT ANY WARRANTY; 
// #			without even the implied warranty of MERCHANTABILITY or 
// #			FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser 
// #			General Public License for more details.
// #
// #    		You should have received a copy of the GNU Lesser 
// #			General Public License along with the StrataCommonTest
// #			Framework. If not, see http://www.gnu.org/licenses/.
// ##########################################################################

package strata1.common.task;

import strata1.common.logger.ILogger;

/****************************************************************************
 * 
 * @author 		
 *     Sapientia Systems
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public 
class TestTask
    extends AbstractTask
{
    private ILogger itsLogger;
    
    /************************************************************************
     * Creates a new {@code TestTask}. 
     *
     * @param name
     */
    public 
    TestTask(String name,Integer taskId,ILogger logger)
    {
        super( name );
        itsLogger = logger;
        setProperty( "taskId",taskId );
    }

    /************************************************************************
     * {@inheritDoc} 
     */
    @Override
    public void 
    execute()
    {
        StringBuilder builder = new StringBuilder();
        
        builder
            .append( "execute()\n" )
            .append( "      name = " )
            .append( getName() ).append( "\n" )
            .append( "    taskId = ")
            .append( getProperty(Integer.class,"taskId")).append( "\n" )
            .append( "    thread = " )
            .append( Thread.currentThread().getName() ).append( "\n" );
        
        itsLogger.logInfo( builder.toString() );
    }

}

// ##########################################################################

File: legacy/csharp/tests/StrataCommonTest/Task/TestTask.cs
//  ##########################################################################
//  # File Name: TestTask.cs
//  # Copyright: 2011-2018, Sapientia Systems, LLC.
//  ##########################################################################
 
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using Strata.Common.Logging;
using Strata.Common.Utility;

namespace Strata.Common.Task
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// $comments$
    /// </summary>
    ///  
    public 
    class TestTask:
        AbstractTask
    {
        public ILogger    Logger { get; set; }
        public AtomicLong Counter { get; set; }

        public
        TestTask(String name,int taskId,ILogger logger,AtomicLong counter):
            base( name )
        {
            SetProperty( "TaskId",taskId );
            Logger = logger;
            Counter = counter;
        }

        public override void
        Execute()
        {
	        StringBuilder builder = new StringBuilder();
	        
	        builder
	            .Append( "Execute()\n" )
	            .Append( "      name = " )
	            .Append( Name ).Append( "\n" )
	            .Append( "    taskId = ")
	            .Append( GetProperty<int>("TaskId")).Append( "\n" )
	            .Append( "    thread = " )
	            .Append( Thread.CurrentThread.ManagedThreadId ).Append( "\n" );
	        
	        Logger.Info( builder.ToString() );
            Counter.IncrementAndGet();
        }
    }
}

//  ##########################################################################
